K2Bridge/KustoDAL/KustoQueryExecutor.cs

```diff
@@ -110,10 +110,31 @@ public async Task<(TimeSpan timeTaken, IDataReader reader)> ExecuteQueryAsync(Qu
                 Logger.LogDebug("[metric] backend query total (sdk) duration: {timeTaken}", timeTaken);
                 return (timeTaken, dataReader);
             }
+            catch (Kusto.Data.Exceptions.SemanticException ex)
+            {
+                Logger.LogError(ex, "Semantic exception - Failed to execute query.");
+
+                // If it was a semantic error AND it is about an invalid field, ignore
+                // the error and return an empty results set, as this is the behaviour of
+                // Kibana
+                if (!ex.Message.Contains(
```

Comment:
I don't like that we have to check the exception text, because that seems like something that might change one day,
but I don't have a better way.

With that being said, I think that rather checking on the entire message (which also contains the query), you should check on `ex.SemanticErrors` or `ex.ErrorMessage`